### ISSUE TITLE
cannot deploy contract

### ISSUE THREAD

**Issue by OliverLaiGHong** [ApeWorX/ape#2087]
### Environment information

- OS: linux (Ubuntu 20.04.6 LTS
- Python Version: 3.11
- `ape` and plugin versions:

```sh
$ ape --version
0.7.19

$ ape plugins list
Installed Plugins
  alchemy      0.7.2
  ens          0.7.1
  etherscan    0.7.2
  foundry      0.7.4
  hardhat      0.7.2
  infura       0.7.3
  polygon      0.7.2
  solidity     0.7.3
  template     0.7.0
  tokens       0.7.1
  vyper        0.7.1
```

- Contents of your `ape-config.yaml` (NOTE: do not post anything private like RPC urls or secrets!):

```sh
$ cat ape-config.yaml
name: play_ape
plugins:
  - name: polygon
  - name: ape-etherscan
  - name: foundry
  
default_ecosystem: polygon

polygon:
  default_network: mainnet
  mainnet:
    default_provider: geth
    
foundry:
  host: auto
  request_timeout: 20
  fork_request_timeout: 600
  fork:
    polygon:
      mainnet:
        upstream_provider: geth
        
geth:
  polygon:
    mainnet:
      uri: http://localhost:8545
```

### What went wrong?

I'm running my own polygon pos full node, and Ape is installed on the same machine. Heimdall, Bor, and Ape are running on their own separate tmux session. Whenever I try to deploy a contract on Anvil of Foundry, a None is returned in some step in the receipt retrieval process, causing an error that prevents the contract deployment. 

No attempt to deploy one on the mainnet has been made yet.

For the command I ran for this, the code that caused the failure, and the full output of the error I received, please see the attached image.
![deployment_tx_always_get_None_as_receipt](https://github.com/ApeWorX/ape/assets/89177119/c759f94a-b316-4cb1-a8dc-919776bef022)

**Comment by OliverLaiGHong** [comment]
I forgot to mention that i have also tried to do it on Alchemy and Ankr, but the same error also occurred.

`ape-config.yaml` difference
- Alchemy:
```
plugins:
  - name: polygon
  - name: ape-etherscan
  - name: foundry
  - name: alchemy

polygon:
  default_network: mainnet
  mainnet:
    default_provider: alchemy
```

- Ankr:
```
polygon:
  default_network: mainnet
  mainnet:
    default_provider: geth

geth:
  polygon:
    mainnet:
      uri: <Ankr URL>
```

**Comment by antazoey** [comment]
This was fixed in 7.20.0. Please try again using an updated version! Please re-open if sill a problem on 7.20 or greater, thank you :)
